### PR TITLE
Domains: Fix the set primary domain notice on success/error

### PR DESCRIPTION
**Pull request by hambai**
#### Changes proposed in this Pull Request

* We're removing the static notice (and the undo button) in favour of standard success notice for consistency with the rest of Calypso

#### Testing instructions

* on the domains list screen click on the "Change primary" button and verify it's working and a standard success notice is shown

### CODE REVIEW

File: client/my-sites/domains/domain-management/list/index.jsx [hunk 1]
```diff
@@ -437,23 +355,25 @@ export class List extends React.Component {
...
-					notice: {
-						type: PRIMARY_DOMAIN_CHANGE_SUCCESS,
-						domainName: domain.name,
-						previousDomainName: currentPrimaryName,
-					},
 				} );
+
+				this.props.successNotice(
+					translate(
+						'Primary Domain changed: All domains will redirect to {{em}}%(domainName)s{{/em}}.',
```
Comment (klimeryk) [inline client/my-sites/domains/domain-management/list/index.jsx]: I know this is from the Asana card, but the capitalization here is bothering me a bit. Maybe let's keep it simple a say `Primary domain changed: all domains will redirect to {{em}}%(domainName)s{{/em}}.`?

File: client/my-sites/domains/domain-management/list/index.jsx [hunk 2]
```diff
@@ -437,23 +355,25 @@ export class List extends React.Component {
...
 					settingPrimaryDomain: false,
 					primaryDomainIndex: currentPrimaryIndex,
-					notice: {
-						type: PRIMARY_DOMAIN_CHANGE_FAIL,
-						domainName: domain.name,
-						error,
-					},
 				} );
+				this.props.errorNotice(
+					translate( "Something went wrong and we couldn't change your Primary Domain." ),
```
Comment (klimeryk) [inline client/my-sites/domains/domain-management/list/index.jsx]: Like above, I'd just keep `primary domain` lowercase - it's not a name of a special product, just a domain that is primary.

File: client/my-sites/domains/domain-management/list/index.jsx (hunk 1 above)
Comment (fditrapani) [reply client/my-sites/domains/domain-management/list/index.jsx]: good suggestion. I like your suggestion Igor!

File: client/my-sites/domains/domain-management/list/index.jsx (hunk 2 above)
Comment (fditrapani) [reply client/my-sites/domains/domain-management/list/index.jsx]: This one too. I prefer it lowercase too.

**Comment by klimeryk** [comment]
Forgot to mention: noticed this misaligned progress indicator (both Firefox and Chromium, Mac, if that makes a difference):
primary

Have not seen it on the list of improvements, so reporting here - but feel free to address in a separate PR.

**Comment by hambai** [comment]
> Have not seen it on the list of improvements, so reporting here - but feel free to address in a separate PR.

Ahh yeah, I noticed that and forgot to report. I'll make a card

File: client/my-sites/domains/domain-management/list/index.jsx [hunk 3]
```diff
@@ -437,23 +355,25 @@ export class List extends React.Component {
...
-					notice: {
-						type: PRIMARY_DOMAIN_CHANGE_SUCCESS,
-						domainName: domain.name,
-						previousDomainName: currentPrimaryName,
-					},
 				} );
+
+				this.props.successNotice(
+					translate(
+						'Primary domain changed: all domains will redirect to {{em}}%(domainName)s{{/em}}.',
```
Comment (a8ci18n) [inline client/my-sites/domains/domain-management/list/index.jsx]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **21** times:
`translate( 'Primary Domain changed: All domains will redirect to {{em}}%(domainName)s{{/em}}.' )` *ES Score: 9*

File: client/my-sites/domains/domain-management/list/index.jsx [hunk 4]
```diff
@@ -437,23 +355,25 @@ export class List extends React.Component {
...
 					settingPrimaryDomain: false,
 					primaryDomainIndex: currentPrimaryIndex,
-					notice: {
-						type: PRIMARY_DOMAIN_CHANGE_FAIL,
-						domainName: domain.name,
-						error,
-					},
 				} );
+				this.props.errorNotice(
+					translate( "Something went wrong and we couldn't change your primary domain." ),
```
Comment (a8ci18n) [inline client/my-sites/domains/domain-management/list/index.jsx]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **21** times:
`translate( 'Something went wrong and we couldn't change your Primary Domain.' )` *ES Score: 12*

**Pull request merged**